packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,539 @@
+    }
+    return this._outputs;
+  }
+
+  /**
+   * Calculates and returns the fee of the transaction in satoshis.
+   * @returns {string} The transaction fee in satoshis
+   */
+  get fee(): string {
+    if (!this._fee) {
+      this._fee = this.calculateFee();
+    }
+    return this._fee.toString();
+  }
+
+  /**
+   * Calculates and returns the fee rate of the transaction in satoshis per vbyte.
+   * @returns {string} The transaction fee rate in satoshis per vbyte
+   */
+  get feeRate(): string {
+    if (!this._feeRate) {
+      this._feeRate = new BigNumber(this.fee).dividedBy(this.vsize);
+    }
+    return this._feeRate.toString();
+  }
+
+  /**
+   * Determines if the transaction is eligible for RBF (Replace-By-Fee).
+   * @returns {boolean} True if the transaction can be replaced using RBF, false otherwise
+   */
+  get canRBF(): boolean {
+    if (this._canRBF === null) {
+      this._canRBF = this.canPerformRBF();
+    }
+    return this._canRBF;
+  }
+
+  /**
+   * Determines if the transaction is eligible for CPFP (Child-Pays-For-Parent).
+   * @returns {boolean} True if CPFP can be performed on this transaction, false otherwise
+   */
+  get canCPFP(): boolean {
+    if (this._canCPFP === null) {
+      this._canCPFP = this.canPerformCPFP();
+    }
+    return this._canCPFP;
+  }
+
+  /**
+   * Recommends the optimal fee bumping strategy based on the current transaction state.
+   * @returns {FeeBumpStrategy} The recommended fee bumping strategy
+   */
+  get recommendedStrategy(): FeeBumpStrategy {
+    if (!this._recommendedStrategy) {
+      this._recommendedStrategy = this.recommendStrategy();
+    }
+    return this._recommendedStrategy;
+  }
+
+  /**
+   * Gets the list of available UTXOs for potential use in fee bumping.
+   * @returns {UTXO[]} An array of available UTXOs
+   */
+  get getAvailableUTXOs(): UTXO[] {
+    return this.availableUTXOs;
+  }
+
+  /**
+   * Gets the current target fee rate in satoshis per vbyte.
+   * @returns {number} The target fee rate
+   */
+  get targetFeeRate(): number {
+    return this._targetFeeRate;
+  }
+
+  /**
+   * Calculates and returns the minimum fee rate required for a successful RBF.
+   * @returns {string} The minimum RBF fee rate in satoshis per vbyte
+   */
+  get rfbFeeRate(): string {
+    if (!this._rbfFeeRate) {
+      const currentFeeRate = this.feeRate;
+
+      const minReplacementFeeRate = BigNumber.max(
+        new BigNumber(currentFeeRate).plus(this.incrementalRelayFee),
+        new BigNumber(this.targetFeeRate),
+      );
+
+      this._rbfFeeRate = minReplacementFeeRate.toString();
+    }
+    return this._rbfFeeRate;
+  }
+
+  /**
+   * Calculates and returns the fee rate required for a successful CPFP.
+   * @returns {string} The CPFP fee rate in satoshis per vbyte
+   */
+  get cpfpFeeRate(): string {
+    if (!this._cpfpFeeRate) {
+      const originalTxVBytes = this.vsize;
+      const packageSize = originalTxVBytes + ESTIMATED_CHILD_TX_SIZE;
```

Comment:
This is going to be problematic. I'm guessing the child tx size was just for P2SH, but what if the package is all P2WSH? The size will be different.

Reply:
added a new util fn `estimateTransactionVsize()` to deal with this inaccurate calculation